### PR TITLE
Add all tabs counter for windows dialog

### PR DESCRIPTION
**Pull request by X1ngyuL1u**
As requested by #8338,
Added a status bar to the windows dialog which shows the total number of tabs and will update along with other elements in the dialog. It will also resize well with the whole window. I had to move the "OK" button a little bit higher in order to add this status bar.

### CODE REVIEW

File: PowerEditor/src/WinControls/WindowsDlg/WindowsDlg.cpp [hunk 1]
```diff
@@ -865,6 +870,19 @@ void WindowsDlg::doClose()
...
+}
+
+//this function will be called everytime when close is performed
+//as well as each time refresh is performed to keep updated
+void WindowsDlg::doCount()
+{
+	TCHAR count[32];
+	wsprintf(count, TEXT("%d"), _idxMap.size());
+	generic_string msg = TEXT("");
+	msg += TEXT("Total: ");
```
Comment (sasumner) [inline PowerEditor/src/WinControls/WindowsDlg/WindowsDlg.cpp:883]: `Total` should be made translatable.

Comment (X1ngyuL1u) [reply PowerEditor/src/WinControls/WindowsDlg/WindowsDlg.cpp]: Updated the total. Now it should be translatable.

**Comment by donho** [comment]
I don't like adding a status bar just for this info.
I think a static text on the bottom of Window dialog to display this info (`Total document number: 25`) will be enough.

**Comment by X1ngyuL1u** [comment]
The original code uses CWinMgr which is hard to manipulate around, plus I think the added status bar can extend spaces for more information in the future if needed, e.g. total number of selected.

**Comment by X1ngyuL1u** [comment]
> I don't like adding a status bar just for this info.
> I think a static text on the bottom of Window dialog to display this info (`Total document number: 25`) will be enough.

I tried to work around with the WinMgr the original code is using, however, I still cannot put he static text to the right place. So this time I tried to put the count to the Title, and now the title of the dialog would be "Windows -- Total tabs opened: XX".

**Review by donho — CHANGES_REQUESTED** [review]
Use Window Caption (title bar) to display info instead of status bar:
"Window (total documents open: 46)"

**Review by donho — APPROVED** [review]

File: PowerEditor/src/WinControls/WindowsDlg/WindowsDlg.cpp [hunk 2]
```diff
@@ -228,7 +229,7 @@ BEGIN_WINDOW_MAP(WindowsDlgMap)
 		BEGINCOLS(WRCT_REST,0,0)                       // Begin list control column
 			BEGINROWS(WRCT_REST,0,0)
 				RCREST(IDC_WINDOWS_LIST)
-				RCSPACE(20)
+				RCSPACE(15)
```
Comment (donho) [inline PowerEditor/src/WinControls/WindowsDlg/WindowsDlg.cpp:232]: Restore `RCSPACE(20)` which is not necessary.

File: PowerEditor/src/WinControls/WindowsDlg/WindowsDlg.cpp [hunk 3]
```diff
@@ -242,6 +243,7 @@ BEGIN_WINDOW_MAP(WindowsDlgMap)
 			RCTOFIT(IDC_WINDOWS_SORT)
 			RCREST(-1)
 			RCTOFIT(IDCANCEL)
+			RCSPACE(14)
```
Comment (donho) [inline PowerEditor/src/WinControls/WindowsDlg/WindowsDlg.cpp:246]: Remove `RCSPACE(14)` which is not necessary.

File: PowerEditor/src/WinControls/WindowsDlg/WindowsDlg.cpp [hunk 4]
```diff
@@ -664,7 +666,7 @@ void WindowsDlg::updateColumnNames()
 
 void WindowsDlg::onSize(UINT nType, int cx, int cy)
 {
-	MyBaseClass::onSize(nType, cx, cy);
+	MyBaseClass::onSize(nType, cx, cy);
```
Comment (donho) [inline PowerEditor/src/WinControls/WindowsDlg/WindowsDlg.cpp:669]: Remove all the space characters after `MyBaseClass::onSize(nType, cx, cy);`.

File: PowerEditor/src/WinControls/WindowsDlg/WindowsDlg.cpp [hunk 5]
```diff
@@ -865,6 +868,22 @@ void WindowsDlg::doClose()
 		}
 		ListView_SetItemCount(_hList, _idxMap.size());
 	}
+	doCount();
```
Comment (donho) [inline PowerEditor/src/WinControls/WindowsDlg/WindowsDlg.cpp:871]: Why do we need to count the document while closing the dialog?

File: PowerEditor/src/WinControls/WindowsDlg/WindowsDlg.cpp [hunk 6]
```diff
@@ -865,6 +868,22 @@ void WindowsDlg::doClose()
...
+//as well as each time refresh is performed to keep updated
+void WindowsDlg::doCount()
+{
+	NativeLangSpeaker* pNativeSpeaker = (NppParameters::getInstance()).getNativeLangSpeaker();
+	
+	TCHAR count[32];
+	wsprintf(count, TEXT("%d"), _idxMap.size());
+	
+	generic_string msg = TEXT("Windows -- ");
+	msg += pNativeSpeaker->getAttrNameStr(TEXT("Total tabs opened: "), WD_ROOTNODE, WD_TABTOTAL);
```
Comment (donho) [inline PowerEditor/src/WinControls/WindowsDlg/WindowsDlg.cpp:884]: Please replace `TEXT("Total tabs opened: ")` by `TEXT("Total documents opened: ")`.

File: PowerEditor/src/WinControls/WindowsDlg/WindowsDlgRc.h [hunk 7]
```diff
@@ -49,4 +49,3 @@
 	#define  IDM_WINDOW_WINDOWS   (IDR_WINDOWS_MENU + 1)
 	#define  IDM_WINDOW_MRU_FIRST (IDR_WINDOWS_MENU + 20)
 	#define  IDM_WINDOW_MRU_LIMIT (IDR_WINDOWS_MENU + 29)
-
```
Comment (donho) [inline PowerEditor/src/WinControls/WindowsDlg/WindowsDlgRc.h:52]: Restore this empty line.

**Review by donho — CHANGES_REQUESTED** [review]
Please take consideration on the remarks.

File: PowerEditor/src/WinControls/WindowsDlg/WindowsDlg.cpp (hunk 2 above)
Comment (X1ngyuL1u) [reply PowerEditor/src/WinControls/WindowsDlg/WindowsDlg.cpp]: Restored

File: PowerEditor/src/WinControls/WindowsDlg/WindowsDlg.cpp (hunk 3 above)
Comment (X1ngyuL1u) [reply PowerEditor/src/WinControls/WindowsDlg/WindowsDlg.cpp]: Removed

File: PowerEditor/src/WinControls/WindowsDlg/WindowsDlg.cpp (hunk 4 above)
Comment (X1ngyuL1u) [reply PowerEditor/src/WinControls/WindowsDlg/WindowsDlg.cpp]: Removed

File: PowerEditor/src/WinControls/WindowsDlg/WindowsDlg.cpp (hunk 5 above)
Comment (X1ngyuL1u) [reply PowerEditor/src/WinControls/WindowsDlg/WindowsDlg.cpp:870]: This is not closing the dialog, this is the button event which closes the chosen tab in the dialog, thus we need to update the count after we closing tabs

File: PowerEditor/src/WinControls/WindowsDlg/WindowsDlg.cpp (hunk 6 above)
Comment (X1ngyuL1u) [reply PowerEditor/src/WinControls/WindowsDlg/WindowsDlg.cpp]: Replaced

File: PowerEditor/src/WinControls/WindowsDlg/WindowsDlgRc.h (hunk 7 above)
Comment (X1ngyuL1u) [reply PowerEditor/src/WinControls/WindowsDlg/WindowsDlgRc.h]: Restored

**Review by donho — APPROVED** [review]

**Pull request closed without merging**